driver-core/src/main/com/mongodb/internal/operation/CreateCollectionOperation.java

```diff
@@ -384,6 +406,9 @@ public void onResult(@Nullable final Void result, @Nullable final Throwable t) {
                 finalCallback.onResult(null, t);
                 return;
             }
+            if (!checkEncryptedFieldsSupported(connection, finalCallback)) {
```

Comment:
The intent would be clearer if this check were moved to line 255.  That would match the behavior of the sync version.

Reply:
I thought moving it closer to the invocation of `getCommandFunctions` (though, not inside the constructor of `ProcessCommandsCallback`) is better. Does not matter much. Done.

Reply:
Notice that `this` is passed as an argument to `executeCommandAsync` at the end of this function, so the check will happen three times if it were left here.  It will give the same answer all three times, but the intent is clearer if it only happens once.

Reply:
Thanks!